extensions/wsc/src/main/java/org/mule/extension/ws/internal/transport/HttpDispatcher.java

```diff
@@ -7,50 +7,105 @@
+   * <p>
+   * Dispatches a Soap message through http adding the SoapAction header, if required, and the content-type.
+   */
+  @Override
+  public WscResponse dispatch(Message message) {
 
     OutputStream os = message.getContent(OutputStream.class);
-    OkHttpClient client = new OkHttpClient();
 
-    String contentType = (String) message.get(Message.CONTENT_TYPE);
-    if (contentType.contains("action")) {
-      // TODO: MULE-11100: MediaType.parse cannot parse a content type that carries the action element.
-      contentType = contentType.substring(0, contentType.indexOf("action")).concat("\"");
+    HttpRequest request = HttpRequest.builder()
+        .setUri(address)
+        .setMethod("POST")
+        .setEntity(new ByteArrayHttpEntity(os.toString().getBytes()))
+        .setHeaders(buildHeaders(message))
+        .build();
+
+    try {
+      HttpResponse response = client.send(request, 5000, false, null);
+      InputStream content = ((InputStreamHttpEntity) response.getEntity()).getInputStream();
+      return new WscResponse(content, response.getHeaderValue(CONTENT_TYPE.toLowerCase()));
+    } catch (IOException e) {
+      throw new ModuleException(e, CANNOT_DISPATCH, "An error occurred while sending the SOAP request");
+    } catch (TimeoutException e) {
+      throw new ModuleException(e, TIMEOUT, "The SOAP request timed out");
     }
+  }
 
-    MediaType mediaType = MediaType.parse(contentType);
-    RequestBody body = RequestBody.create(mediaType, os.toString());
-    Request.Builder request = new Request.Builder()
-        .url(address)
-        .post(body)
-        .addHeader("cache-control", "no-cache");
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void dispose() {
+    client.stop();
+  }
+
+  private ParameterMap buildHeaders(Message message) {
+    ParameterMap headers = new ParameterMap();
+    headers.put(CONTENT_TYPE, (String) message.get(CONTENT_TYPE));
 
+    String soapAction = (String) message.getExchange().get(MULE_SOAP_ACTION);
     if (soapAction != null) {
-      request.addHeader("SOAPAction", soapAction);
+      headers.put("SOAPAction", soapAction);
     }
+    return headers;
+  }
 
-    try {
-      return client.newCall(request.build()).execute();
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
+  /**
+   * Creates a default {@link HttpDispatcher} using the provided {@link HttpService} and with a fixed set of attributes.
+   *
+   * @param address the address we want to dispatch http messages to.
+   * @param httpService the configured {@link HttpService} used to create the http client.
+   * @return a new {@link HttpDispatcher} default instance.
+   */
+  public static HttpDispatcher createDefault(String address, HttpService httpService) {
+    HttpClientFactory clientFactory = httpService.getClientFactory();
+    HttpClient client = clientFactory.create(new HttpClientConfiguration.Builder()
+        .setThreadNamePrefix("wsc")
+        .setOwnerName("wsc")
```

Comment:
If each config will have one of this then it may be a good idea to a have a custom name that addresses that. It will make things easier for troubleshooting in the future.